dataplane/api-server/src/lib.rs

```diff
@@ -5,33 +5,128 @@ SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
 */
 
 pub mod backends;
+pub mod config;
 pub mod netutils;
 pub mod server;
 
-use std::net::{Ipv4Addr, SocketAddrV4};
+use std::{
+    fs,
+    net::{Ipv4Addr, SocketAddrV4},
+};
 
-use anyhow::Error;
+use anyhow::{Context, Result};
 use aya::maps::{HashMap, MapData};
-use tonic::transport::Server;
+use log::info;
+use tonic::transport::{Certificate, Identity, Server, ServerTlsConfig};
 
 use backends::backends_server::BackendsServer;
 use common::{BackendKey, BackendList, ClientKey, LoadBalancerMapping};
+use config::TLSConfig;
 
 pub async fn start(
     addr: Ipv4Addr,
     port: u16,
     backends_map: HashMap<MapData, BackendKey, BackendList>,
     gateway_indexes_map: HashMap<MapData, BackendKey, u16>,
     tcp_conns_map: HashMap<MapData, ClientKey, LoadBalancerMapping>,
-) -> Result<(), Error> {
-    let (_, health_service) = tonic_health::server::health_reporter();
-
-    let server = server::BackendService::new(backends_map, gateway_indexes_map, tcp_conns_map);
-    // TODO: mTLS https://github.com/Kong/blixt/issues/50
-    Server::builder()
-        .add_service(health_service)
-        .add_service(BackendsServer::new(server))
-        .serve(SocketAddrV4::new(addr, port).into())
-        .await?;
+    tls_config: Option<TLSConfig>,
+) -> Result<()> {
+    // Tonic itself doesn't provide a built-in mechanism for selectively
+    // applying TLS based on routes, as TLS configuration is tied to the
+    // entire server and managed at the transport layer, not at the
+    // application layer where routes are defined.
+    //
+    // Solution: separate gRPC services
+    //
+    // Public server without mTLS for healthchecks
```

Comment:
If it's OK with you I would like to put a finer point on this comment to make it clear that no dataplane functionality is enabled through this unencrypted endpoint:

```suggestion
    // Public server without TLS (healthchecks ONLY)
```